client_linux.go

```diff
@@ -1046,6 +1265,82 @@ func parseRings(msgs []genetlink.Message) ([]*Rings, error) {
 	return rings, nil
 }
 
+// parseFeatyres parses FeatureInfo structures from a slice of generic netlink
+// messages.
+func parseFeatures(msgs []genetlink.Message, featureStrings StringSet) ([]FeatureInfo, error) {
+	if len(msgs) != 1 {
+		return nil, fmt.Errorf("ethtool: unexpected number of Features messages: %d", len(msgs))
+	}
+
+	m := msgs[0]
+
+	ad, err := netlink.NewAttributeDecoder(m.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	var hw, wanted, active, noChange bitset
+
+	for ad.Next() {
+		switch ad.Type() {
+		case unix.ETHTOOL_A_FEATURES_HW:
+			ad.Nested(func(nad *netlink.AttributeDecoder) error {
+				hw, err = newBitset(nad)
+
+				return err
+			})
+		case unix.ETHTOOL_A_FEATURES_WANTED:
+			ad.Nested(func(nad *netlink.AttributeDecoder) error {
+				wanted, err = newBitset(nad)
+
+				return err
+			})
+		case unix.ETHTOOL_A_FEATURES_ACTIVE:
+			ad.Nested(func(nad *netlink.AttributeDecoder) error {
+				active, err = newBitset(nad)
+
+				return err
+			})
+		case unix.ETHTOOL_A_FEATURES_NOCHANGE:
+			ad.Nested(func(nad *netlink.AttributeDecoder) error {
+				noChange, err = newBitset(nad)
+
+				return err
+			})
+		}
+	}
+
+	if err := ad.Err(); err != nil {
+		return nil, err
+	}
+
+	if len(hw) != len(wanted) || len(wanted) != len(active) || len(active) != len(noChange) {
+		return nil, fmt.Errorf("ethtool: unexpected number of feature bitsets: %d, %d, %d, %d",
+			len(hw), len(wanted), len(active), len(noChange))
+	}
+
+	numBits := len(hw) * 32
```

Comment:
constify thie size?

Reply:
you mean like `sizeof(uint32)`? it's the underlying type of `bitSet`, that is `[]uint32`.